base/path.jl

```diff
@@ -611,3 +611,21 @@ relpath(path::AbstractString, startpath::AbstractString) =
 for f in (:isdirpath, :splitdir, :splitdrive, :splitext, :normpath, :abspath)
     @eval $f(path::AbstractString) = $f(String(path))
 end
+
+"""
+    uripath(path::AbstractString)
+
+Encode `path` as a URI as per RFC1738, RFC3986, and the
+[Freedesktop File URI spec](https://www.freedesktop.org/wiki/Specifications/file-uri-spec/).
+"""
+function uripath(path::AbstractString)
```

Comment:
```suggestion
uripath(path::AbstractString) = String(path)
function uripath(path::String)
```
(since this is explicitly operating on the UTF-8 represention, it may be best to enforce that upfront)

Reply:
I think I get where you were going here, even if what's shown here isn't quite right.